reconcile/change_owners/changes.py

```diff
@@ -454,3 +457,35 @@ def parse_bundle_changes(
             )
 
     return change_list
+
+
+def aggregate_resource_changes(
+    bundle_changes: list[BundleFileChange],
+    content_store: list[dict[str, Any]],
+) -> list[BundleFileChange]:
+    new_bundle_changes: list[BundleFileChange] = []
+    resource_changes: list[BundleFileChange] = []
+    for change in bundle_changes:
+        new_bundle_changes.append(change)
+        for file_ref in change.old_backrefs | change.new_backrefs:
+            file_content = next(c for c in content_store if c["path"] == file_ref.path)
```

Comment:
to speed up `file_content` lookup, can pre build a lookup dict

```python
file_content_by_path = {c["path"]: c for c in content_store}
```

then the final logic can be like

```python
    file_content_by_path = {c["path"]: c for c in content_store}
    resource_changes = [
        BundleFileChange(
            fileref=change.fileref,
            old=file_content,
            new=file_content,
            old_content_sha="",
            new_content_sha="",
            diffs=[
                Diff(
                    path=jsonpath_ng.parse(file_ref.json_path),
                    diff_type=DiffType.CHANGED,
                    old=file_content,
                    new=file_content,
                )
            ],
        )
        for change in bundle_changes
        for file_ref in change.old_backrefs | change.new_backrefs
        if (file_content := file_content_by_path.get(file_ref.path))
    ]
    return bundle_changes + resource_changes
```

Reply:
i wanted to remain ordered